nightfall-deployer/contracts/Shield.sol

```diff
@@ -185,7 +185,7 @@ contract Shield is Stateful, Structures, Config, Key_Registry {
           uint256(t.value)
         );
     } else if(t.tokenType == TokenType.ERC721) {
-      if (t.value != 0 || t.tokenId == ZERO) // value should always be equal to 0 and tokenId cannot be equal to ZERO
```

Comment:
Without this change, do the tests fail somewhere?

Reply:
Shield contract was incorrect. An ERC721 can have id 0

Reply:
That is true according to [EIP](https://eips.ethereum.org/EIPS/eip-721)
> While some ERC-721 smart contracts may find it convenient to start with ID 0 and simply increment by one for each new NFT, callers SHALL NOT assume that ID numbers have any specific pattern to them, and MUST treat the ID as a “black box”.

On other hand value does not exist in ERC721. It is essentially a null field. Because solidity does not have null as a concept, we use 0. There can't be any value including 1 for ERC721 because value does not represent count. 

Can we change this check to  `if (t.value != 0)`?

Reply:
ok

Reply:
If we need balance for ERC721 then we need to change how we calculate balance after querying the database in Client. This will be different from that of ERC20 . The shield contract should not be written to cater for this. 
The logic to get balance of ERC721 would be to count ERC721 tokens from database that are not `onChain` and not `isNullified`?

Reply:
as agreed, i maintain check of erc720 value to 0 and disable erc720 tests. There will be a new issue to describe the problem and track the solution